src/common/uilocale.cpp

```diff
@@ -275,6 +275,330 @@ wxLocaleIdent wxLocaleIdent::FromTag(const wxString& tag)
+        }
+        return locId;
+    }
+}
+
+/* static */
+wxLocaleIdent wxLocaleIdent::RemoveLikelySubtags(const wxLocaleIdent& localeIdent, int subtagsFavor)
+{
+    // Extend input with likely subtags
+    wxLocaleIdent locId = AddLikelySubtags(localeIdent);
+
+    // If extending failed, return
+    if (locId.IsEmpty())
+    {
+        return locId;
+    }
+
+    // Get the components language, script, region
+    wxString language = locId.GetLanguage();
+    wxString script = locId.GetScript();
+    wxString region = locId.GetRegion();
+
+    wxLocaleIdent trial = localeIdent;
+    trial.Language(language);
+    trial.Script("");
+    trial.Region("");
+
+    // Test "language"
+    locId = AddLikelySubtags(trial);
+    if (script.IsSameAs(locId.GetScript()) && region.IsSameAs(locId.GetRegion()))
+    {
+        return trial;
+    }
+
+    // Test "language-region" if favoring region or
+    //      "language-script" if favoring script
+    if (subtagsFavor == wxSubtags_FavorRegion)
+    {
+        trial.Region(region);
+    }
+    else
+    {
+        trial.Script(script);
+    }
+    locId = AddLikelySubtags(trial);
+    if (script.IsSameAs(locId.GetScript()) && region.IsSameAs(locId.GetRegion()))
+    {
+        return trial;
+    }
+
+    // Test "language-script" if favoring region or
+    //      "language-region" if favoring script
+    if (subtagsFavor != wxSubtags_FavorRegion)
+    {
+        trial.Region(region);
+    }
+    else
+    {
+        trial.Script(script);
+    }
+    locId = AddLikelySubtags(trial);
+    if (script.IsSameAs(locId.GetScript()) && region.IsSameAs(locId.GetRegion()))
+    {
+        return trial;
+    }
+
+    // No match, return full identifier
+    return locId;
+}
+
+// "Infinity" value needs to be larger than any real language distance
+static const int DISTANCE_INFINITY = 1000;
+
+/* static */
+wxString wxLocaleIdent::GetBestMatch(const wxVector<wxString>& desired, const wxVector<wxString>& supported)
+{
+    // Default distance values
+    static int defaultLanguageDistance = wxUILocaleImpl::GetMatchDistance("*", "*");
+    static int defaultScriptDistance = wxUILocaleImpl::GetMatchDistance("*-*", "*-*");
+    static int defaultRegionDistance = wxUILocaleImpl::GetMatchDistance("*-*-*", "*-*-*") + 1;
+    static int defaultRegionGroupDistance = wxUILocaleImpl::GetMatchDistance("*-*-*", "*-*-*");
+    static int defaultDemotion = wxUILocaleImpl::GetMatchDistance("en-*-*", "en-*-*") + 1;
+    static int threshold = defaultScriptDistance;
+
+    // Maximized locale identifiers associated with the given desired and supported locale tags
+    wxLocaleIdent desiredLocaleIn;
+    wxLocaleIdent desiredLocaleMax;
+    wxLocaleIdent supportedLocaleIn;
+    std::vector<wxLocaleIdent> supportedLocalesIn;
+    std::vector<wxLocaleIdent> supportedLocalesMax;
+    std::vector<int> supportedLocalesMinDistance;
+
+    // Indexes of best match
+    int bestDesired = -1;
+    wxUnusedVar(bestDesired);
```

Comment:
Well, for one, without `bestDesired` it is difficult to debug the code. Secondly, it may be of interest to which of the given desired languages the selected supported language actually matches, although this information is currently not passed onward to the calling method.